jupyterhub_singleuser_profiles/profiles.py

```diff
@@ -13,20 +13,30 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+_MAIN_CONFIG_MAP_NAME = "jupyter-singleuser-profiles"
+_PROFILE_CONFIG_MAP_KEY = "jupyterhub-singleuser-profiles.yaml"
 _JUPYTERHUB_USER_NAME_ENV = "JUPYTERHUB_USER_NAME"
 _USER_CONFIG_MAP_TEMPLATE = "jupyterhub-singleuser-profile-%s"
 _USER_CONFIG_PROFILE_NAME = "@singleuser@"
+_JSP_CONFIG_MAP_LABEL_NAME = "jupyterhub.opendatahub.io/profile-type"
+_USER_CONFIG_MAP_LABEL_VALUE = "user"
+_PROFILE_CONFIG_MAP_LABEL_VALUE = "profile"
+_JSP_USER_LABEL_NAME = "jupyterhub.opendatahub.io/user"
+
+_SINGLE_USER_PROFILE_CM_LABEL = "%s=%s" % (_JSP_CONFIG_MAP_LABEL_NAME, _PROFILE_CONFIG_MAP_LABEL_VALUE)
 
 class SingleuserProfiles(object):
   GPU_MODE_SELINUX = "selinux"
   GPU_MODE_PRIVILEGED = "privileged"
-  def __init__(self, server_url, token, namespace=None, verify_ssl=True, gpu_mode=None):
+  def __init__(self, server_url, token, namespace=None, verify_ssl=True, gpu_mode=None, cm_profile_label = None):
     self.profiles = []
     self.service = Service(server_url, token, namespace, verify_ssl)
     self.api_client = None
     self.namespace = namespace #TODO why do I need to pass namespace?
     self.gpu_mode = gpu_mode
     self.oapi_client = None
+    if profile_label:
```

Comment:
@accorvin This is going to influence config maps deployed in IDH, but I have added a parameter for overriding a label to offer some backwards compatibility - WDYT?